packages/core/src/node/messaging/messaging-contribution.ts

```diff
@@ -176,29 +125,33 @@ export class MessagingContribution implements BackendApplicationContribution, Me
         return connectionContainer;
     }
 
-    protected getConnectionChannelHandlers(socket: Socket): MessagingContribution.ConnectionHandlers<WebSocketChannel> {
+    protected getConnectionChannelHandlers(socket: Socket): MessagingContribution.ConnectionHandlers<Channel> {
         const connectionContainer = this.createSocketContainer(socket);
         bindContributionProvider(connectionContainer, ConnectionHandler);
         connectionContainer.load(...this.connectionModules.getContributions());
         const connectionChannelHandlers = new MessagingContribution.ConnectionHandlers(this.channelHandlers);
         const connectionHandlers = connectionContainer.getNamed<ContributionProvider<ConnectionHandler>>(ContributionProvider, ConnectionHandler);
         for (const connectionHandler of connectionHandlers.getContributions(true)) {
             connectionChannelHandlers.push(connectionHandler.path, (_, channel) => {
-                const connection = createWebSocketConnection(channel, new ConsoleLogger());
-                connectionHandler.onConnection(connection);
+                connectionHandler.onConnection(channel);
             });
         }
         return connectionChannelHandlers;
     }
 
-    protected createChannel(id: number, socket: Socket): WebSocketChannel {
-        return new WebSocketChannel(id, content => {
-            if (socket.connected) {
-                socket.send(content);
-            }
-        });
-    }
+}
 
+function toIWebSocket(socket: Socket): IWebSocket {
+    return {
+        close: () => {
+            socket.disconnect();
+        },
+        isConnected: () => socket.connected,
+        onClose: cb => socket.on('disconnect', reason => cb(reason)),
+        onError: cb => socket.on('error', error => cb(error)),
+        onMessage: cb => socket.on('message', data => cb(toArrayBuffer(data))),
```

Comment:
I don't believe that **all** communications should be considered as array buffers.

If I understand correctly we only use those optimized buffers for RPC communication, so we should only expect and handle array buffer when running this new RPC protocol?